### PR TITLE
Add explanation of how Vim machine works in Readme

### PR DESCRIPTION
**Pull request by IamTheFij**
How's this? Too long?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -36,7 +36,72 @@ To run this problem on a python Turing machine, just call `make run`. To run
...
+current cursor to the end of the line and then executes the default register as
+a macro. This allows us to encode motions in lines and then execute them. We
+then chain lines together by ending lines with moving to a mark, or a search
+result, and then yanking and executing that line.
+
+Using that nifty trick, we begin by yanking the first line and executing it.
+That then sets off our mark initialization. We then jump to various markers in
+the file and then mark those positions using `m<letter>` where `<letter>` is
+whatever we want to call that mark. Generally the first initial of whatever the
+thing we're marking is. Once everything is marked, we then begin the state
```
Comment (ealter) [inline README.md]: I'd just say, "we search for _<letter>: in the code and then use that to mark the pointer".

File: README.md [hunk 2]
```diff
@@ -36,7 +36,72 @@ To run this problem on a python Turing machine, just call `make run`. To run
...
+transitions.
+
+We begin a state transition by executing a long command (located at `_n:`)
+which jumps to the tape marker, yanks it, then jumps to the current state
+marker, yanks that too, and then searches for some transition that contains both
+the state and tape value. Once it gets to that line, it jumps to the command
```
Comment (ealter) [inline README.md]: Nit: s/value/values

File: README.md [hunk 3]
```diff
@@ -36,7 +36,72 @@ To run this problem on a python Turing machine, just call `make run`. To run
...
+to the tape or state marks (`` `t `` or `` `k `` respectively) and then using
+`cw` or `C` to change the value. We then move the pointer by jumping to the tape
+position (`` `t ``) and then moving a word forward (`W`) or backward (`B`), and
+then marking the new tape position.
+
+The last real piece of complication is extending the tape. We're living in a
+world with unlimited tapes! What a time to be alive! This is done through a
+series of nifty hacks. First, we have a modeline that sets `whichwrap+b,s`. This
+allows us to move across line breaks and keep the tape all in the screen. Next,
+the line directly under the tape contains a "fake" value that , when added to
```
Comment (ealter) [inline README.md]: Nit: extra space between 'that' and ','

File: README.md [hunk 4]
```diff
@@ -36,7 +36,72 @@ To run this problem on a python Turing machine, just call `make run`. To run
 this problem on a Vim Turing machine, just call `make run-vim-machine`. To see
 the Vim Turing machine without running it, call `make open-vim-machine`.
 
+So Vim did what? Wait. How does does it even?
```
Comment (ealter) [inline README.md]: s/does does/does

Comment (IamTheFij) [reply README.md]: Fixed.

File: README.md (hunk 1 above)
Comment (IamTheFij) [reply README.md]: Updated.

File: README.md (hunk 2 above)
Comment (IamTheFij) [reply README.md]: Fixed.

File: README.md (hunk 3 above)
Comment (IamTheFij) [reply README.md]: Fixed.

**Pull request merged**